pkg/tfbridge/provider_test.go

```diff
@@ -3259,3 +3259,181 @@ func TestSchemaFuncsNotCalledDuringRuntime(t *testing.T) {
+				"testprov_testres": {
+					Schema: map[string]*schema.Schema{
+						"x": {
+							Type: schema.TypeMap,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{},
+							},
+							MaxItems: 1,
+						},
+					},
+					Timeouts: &schema.ResourceTimeout{
+						Default: tc.schemaTimeout,
+						Create:  tc.schemaTimeout,
+						Update:  tc.schemaTimeout,
+						Delete:  tc.schemaTimeout,
+					},
+					CreateContext: func(
+						ctx context.Context, rd *schema.ResourceData, i interface{},
+					) diag.Diagnostics {
+						t := rd.Timeout(schema.TimeoutCreate)
+						capturedTimeout = &t
+						rd.SetId(id)
+						return diag.Diagnostics{}
+					},
+					UpdateContext: func(
+						ctx context.Context, rd *schema.ResourceData, i interface{},
+					) diag.Diagnostics {
+						t := rd.Timeout(schema.TimeoutUpdate)
+						capturedTimeout = &t
+						return diag.Diagnostics{}
+					},
+					DeleteContext: func(
+						ctx context.Context, rd *schema.ResourceData, i interface{},
+					) diag.Diagnostics {
+						t := rd.Timeout(schema.TimeoutDelete)
+						capturedTimeout = &t
+						return diag.Diagnostics{}
+					},
+				},
+			},
+		}
+
+		providerInfo := ProviderInfo{
+			Name: "testprov",
+			Resources: map[string]*ResourceInfo{
+				"testprov_testres": {
+					Tok: tokens.Type(tok),
+				},
+			},
+		}
+
+		shimmedProvider := shimv2.NewProvider(upstreamProvider)
+
+		bridgedProvider := &Provider{
+			tf:   shimmedProvider,
+			info: providerInfo,
+		}
+
+		bridgedProvider.initResourceMaps()
+
+		switch tc.cud {
+		case "Create":
+			_, err := bridgedProvider.Create(context.Background(), &pulumirpc.CreateRequest{
+				Urn:        urn,
+				Properties: &structpb.Struct{},
+				Timeout:    seconds(tc.userSpecifiedTimeout),
+			})
+			require.NoError(t, err)
+		case "Update":
+			_, err := bridgedProvider.Update(context.Background(), &pulumirpc.UpdateRequest{
+				Id:      id,
+				Urn:     urn,
+				Olds:    &structpb.Struct{},
+				News:    &structpb.Struct{},
+				Timeout: seconds(tc.userSpecifiedTimeout),
+			})
+			require.NoError(t, err)
+		case "Delete":
+			_, err := bridgedProvider.Delete(context.Background(), &pulumirpc.DeleteRequest{
+				Id:         id,
+				Urn:        urn,
+				Properties: &structpb.Struct{},
+				Timeout:    seconds(tc.userSpecifiedTimeout),
+			})
+			require.NoError(t, err)
+		}
+
+		return capturedTimeout
+	}
+
+	expectedTimeout := func(tc testCase) time.Duration {
+		if tc.schemaTimeout == nil && tc.userSpecifiedTimeout == nil {
+			return 20 * time.Minute
+		}
+		if tc.userSpecifiedTimeout != nil {
+			return *tc.userSpecifiedTimeout
+		}
+		return *tc.schemaTimeout
+	}
+
+	skips := func(tc testCase) string {
+		if tc.userSpecifiedTimeout != nil {
+			return ""
+		}
+		if tc.schemaTimeout == nil {
+			return ""
+		}
+		switch tc.cud {
+		case "Update", "Delete":
+			return "TODO[pulumi/pulumi-terraform-bridge#1651] - schema-specified timeouts are not " +
```

Comment:
This suspicion is now confirmed for #1651  - custom Delete or Update timeouts from the upstream provider schema are not currently respected. Any chance that's by design? If we want it respected I can follow up with a quick PR here.

Reply:
Unless we can find a reason why they are not respected, let's respect them.

This is less Chesterton's fence then a pile of scattered sticks in a field.